admin-api-server/src/nft/nft.service.ts

```diff
@@ -5,17 +5,37 @@ import {
   Injectable,
   Inject,
 } from '@nestjs/common';
-import { NFT_IMAGE_PREFIX, PG_CONNECTION } from 'src/constants';
+import { PG_CONNECTION } from 'src/constants';
 import { DbPool } from 'src/db.module';
 import { STMResultStatus, StateTransitionMachine, Actor } from 'roles_stm';
 import { User } from 'src/user/entities/user.entity';
 import { NftEntity } from './entities/nft.entity';
 import { RoleService } from 'src/role/role.service';
 import { S3Service } from './s3.service';
-import { NftFilterParams } from './params';
 import { Lock } from 'async-await-mutex-lock';
+import { QueryParams } from 'src/types';
+import { convertToSnakeCase, prepareFilterClause, prepareNftFilterClause } from 'src/utils';
+import { UpdateNft } from './nft.controller';
 const fs = require('fs');
 
+const getSelectStatement = (
+  whereClause = '',
+  limitClause = '',
+  sortField = 'id',
+  sortDirection = 'ASC',
+): string => `SELECT id, state, created_by, created_at, updated_at, json_agg(json_build_object('name', nftattr.name, 'value', nftattr.value)) as attributes
+FROM nft
+INNER JOIN nft_attribute nftattr on nftattr.nft_id = nft.id ${whereClause}
+GROUP BY nft.id
+ORDER BY ${sortField} ${sortDirection} ${limitClause}`;
+
+const getSelectCountStatement = (
+  whereClause = '',
+): string => `SELECT COUNT(id) FROM nft
+       inner join nft_attribute nftattr on nftattr.nft_id = nft.id ${whereClause}
+       GROUP BY nft.id
+       ORDER BY $1`;
```

Comment:
There is a bug in here. Because if `whereClause` is defined, it already takes bind parameter `$1`. In which case `order by ` shouldn't be using `$1` but some later `$`.